zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/UserTaskQueryControllerTest.java

```diff
@@ -0,0 +1,393 @@
+        .thenReturn(userTaskServices);
+  }
+
+  @Test
+  void shouldSearchUserTasksWithEmptyBody() {
+    // given
+    when(userTaskServices.search(any(UserTaskQuery.class))).thenReturn(SEARCH_QUERY_RESULT);
+    // when / then
+    webClient
+        .post()
+        .uri(USER_TASKS_SEARCH_URL)
+        .exchange()
+        .expectStatus()
+        .isOk()
+        .expectHeader()
+        .contentType(MediaType.APPLICATION_JSON)
+        .expectBody()
+        .json(EXPECTED_SEARCH_RESPONSE);
+
+    verify(userTaskServices).search(new UserTaskQuery.Builder().build());
+  }
+
+  @Test
+  void shouldSearchUserTasksWithEmptyQuery() {
+    // given
+    when(userTaskServices.search(any(UserTaskQuery.class))).thenReturn(SEARCH_QUERY_RESULT);
+    final String request = "{}";
+    // when / then
+    webClient
+        .post()
+        .uri(USER_TASKS_SEARCH_URL)
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(request)
+        .exchange()
+        .expectStatus()
+        .isOk()
+        .expectHeader()
+        .contentType(MediaType.APPLICATION_JSON)
+        .expectBody()
+        .json(EXPECTED_SEARCH_RESPONSE);
+
+    verify(userTaskServices).search(new UserTaskQuery.Builder().build());
+  }
+
+  @Test
+  void shouldSearchUserTasksWithSorting() {
+    // given
+    when(userTaskServices.search(any(UserTaskQuery.class))).thenReturn(SEARCH_QUERY_RESULT);
+    final var request =
+        """
+        {
+            "sort": [
+                {
+                    "field": "creationTime",
+                    "order": "desc"
+                },
+                {
+                    "field": "completionTime",
+                    "order": "asc"
+                }
+            ]
+        }""";
+    // when / then
+    webClient
+        .post()
+        .uri(USER_TASKS_SEARCH_URL)
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(request)
+        .exchange()
+        .expectStatus()
+        .isOk()
+        .expectHeader()
+        .contentType(MediaType.APPLICATION_JSON)
+        .expectBody()
+        .json(EXPECTED_SEARCH_RESPONSE);
+
+    verify(userTaskServices)
+        .search(
+            new UserTaskQuery.Builder()
+                .sort(
+                    new UserTaskSort.Builder().creationDate().desc().completionDate().asc().build())
+                .build());
+  }
+
+  @Test
+  void shouldInvalidateUserTasksSearchQueryWithBadSortOrder() {
+    // given
+    final var request =
+        """
+        {
+            "sort": [
+                {
+                    "field": "creationTime",
+                    "order": "dsc"
+                }
+            ]
+        }""";
+    final var expectedResponse =
+        String.format(
+            """
+        {
+          "type": "about:blank",
+          "title": "Bad Request",
+          "status": 400,
+          "detail": "Unknown sortOrder: dsc",
```

Comment:
❌ Heads-up: This will need to change after you rebased your PR since we aligned the problem response handling.
```suggestion
          "type": "about:blank",
          "title": "INVALID_ARGUMENT",
          "status": 400,
          "detail": "Unknown sortOrder: dsc.",
```